framework/test/gtest/ResourceCompilerTest.cpp

```diff
@@ -448,12 +548,74 @@ void testBundleManifestZipAdd(const std::string& rcbinpath,
   testExists(entryNames, "mybundle/resource1/");
 }
 
+/*
+ * Use resource compiler to create zip files with various compression levels
+ *
+ * Working directory is changed temporarily to tempdir because of --res-add option
+ */
+TEST_F(ResourceCompilerTest, testCompressionLevel)
+{
+  std::ostringstream cmd;
+  cmd << rcbinpath;
+  cmd << " --bundle-name mybundle ";
+  cmd << " --manifest-add manifest.json ";
+  cmd << " --out-file ExampleCompressionLevel0.zip ";
+  cmd << " --res-add resource1/resource1.txt ";
+  cmd << " --zip-add tomerge.zip";
+  cmd << " --compression-level 0";
+
+  auto cwdir = util::GetCurrentWorkingDirectory();
+  ChangeDirectory(tempdir);
+  //Test that --compression-level = 0 successfully creates a zip file.
+  ASSERT_EQ(EXIT_SUCCESS, runExecutable(cmd.str()));
+  ChangeDirectory(cwdir);
+
+  ZipFile zip(tempdir + "ExampleCompressionLevel0.zip");
+  //Check number of entries of zip.
+  ASSERT_EQ(zip.size(), 6);
+
+  auto entryNames = zip.getNames();
+  testExists(entryNames, "mybundle/manifest.json");
+  testExists(entryNames, "mybundle/");
+  testExists(entryNames, "mybundle/resource1/resource1.txt");
+  testExists(entryNames, "mybundle/resource1/");
+  testExists(entryNames, "mybundle/resource2/resource2.txt");
+  testExists(entryNames, "mybundle/resource2/");
+
+  cmd.str("");
+  cmd.clear();
```

Comment:
Are they doing the same thing?

Reply:
How about cmd.str(std::string());

Reply:
yes changed it to cmd.str(std::string());